overlord/snapstate/snapstate.go

```diff
@@ -422,21 +422,38 @@ func Update(s *state.State, name, channel string, userID int, flags Flags) (*sta
 		channel = snapst.Channel
 	}
 
-	updateInfo, err := updateInfo(s, &snapst, channel, userID, flags)
-	if err != nil {
-		return nil, err
+	ss := &SnapSetup{
+		Channel: channel,
+		UserID:  userID,
+		Flags:   SnapSetupFlags(flags),
 	}
-	if err := checkRevisionIsNew(name, &snapst, updateInfo.Revision); err != nil {
+
+	var info *snap.Info
+	if revision.Unset() {
+		// good ol' refresh
+		info, err = updateInfo(s, &snapst, channel, userID, flags)
+	} else {
+		var sideInfo *snap.SideInfo
+		for _, si := range snapst.Sequence {
+			if si.Revision == revision {
+				sideInfo = si
+				break
+			}
+		}
+		if sideInfo == nil {
+			// refresh from given revision from store
+			info, err = snapInfo(s, name, channel, revision, userID, flags)
+		} else {
+			// refresh-to-local
+			info, err = readInfo(name, sideInfo)
```

Comment:
this is the dangerous case, right? if we still bailed out here this would be mergeable alone?